mux/test/muxer.go

```diff
@@ -0,0 +1,541 @@
+package test
```

Comment:
I have a feeling that having `muxer_test.go` is better, rather than having separate test folder

Reply:
These are conformance tests _for muxers_, rather than tests _of muxers_. The mux adaptors call out to these to test themselves, e.g. https://github.com/whyrusleeping/go-smux-yamux/blob/master/yamux_test.go